libraries/app.rb

```diff
@@ -80,23 +80,43 @@ def install_app
         end
 
         def install_dmg
+          temp_pkg_dir = ::File.join(Chef::Config[:file_cache_path], "Xcode_#{new_resource.version}")
+          temp_pkg_file = ::File.join(temp_pkg_dir, ::File.basename(new_resource.url))
+
+          directory temp_pkg_dir do
+            recursive true
+          end
+
           dmg_package new_resource.app do
             source new_resource.url
             checksum new_resource.checksum
             dmg_name ::File.basename(new_resource.url, '.dmg')
             owner 'root'
             type 'app'
+            destination temp_pkg_dir
             action :install
           end
+
+            execute "mv -f #{temp_pkg_dir}/Xcode.app #{install_dir}" do
```

Comment:
This does work and actually runs after the `dmg_package`, unlike the commented option below.